Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveE.cpp

```diff
@@ -193,15 +194,14 @@ void FiniteDifferenceSolver::MacroscopicEvolveECartesian (
             },
 
             [=] AMREX_GPU_DEVICE (int i, int j, int k){
-#ifdef AMREX_USE_EB
 #ifdef WARPX_DIM_3D
-                if (ly(i,j,k) <= 0) return;
+                if (ly && ly(i,j,k) <= 0) { return; }
 #elif defined(WARPX_DIM_XZ)
                 //In XZ Ey is associated with a mesh node, so we need to check if the mesh node is covered
                 amrex::ignore_unused(ly);
```

Comment:
Could move this line outside the device lambda.

Reply:
done (in `#ifdef WARPX_DIM_XZ`)